crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -151,18 +112,56 @@ fn get_invocation_status<S: StorageAccess>(
     // Try read the old one first
     // The underlying assumption is that an invocation status will never exist in both old and new
     // invocation status table.
-    if let Some(s) =
-        storage.get_value::<_, InvocationStatus>(create_invocation_status_key(invocation_id))?
+    if let Some(s) = storage
+        .get_value::<_, InvocationStatusV1>(create_invocation_status_key_v1(invocation_id))?
     {
-        return Ok(s);
+        return Ok(s.0);
     }
 
     // todo: Read first from new invocation status table once the runtime writes to this table
     storage
-        .get_value::<_, InvocationStatusV2>(create_neo_invocation_status_key(invocation_id))
+        .get_value::<_, InvocationStatus>(create_invocation_status_key(invocation_id))
+        .map(|value| {
+            if let Some(invocation_status) = value {
+                invocation_status
+            } else {
+                InvocationStatus::Free
+            }
+        })
+}
+
+pub async fn try_migrate_on_read(
+    storage: &mut PartitionStoreTransaction<'_>,
+    invocation_id: &InvocationId,
+) -> Result<Option<InvocationStatus>> {
+    let v1_key = create_invocation_status_key_v1(invocation_id);
+    if let Some(status_v1) = storage.get_value::<_, InvocationStatusV1>(v1_key.clone())? {
+        trace!("Migrating invocation {invocation_id} from InvocationStatus V1 to V2");
+        put_invocation_status(
+            storage,
+            &InvocationId::from_parts(
+                *v1_key.partition_key_ok_or()?,
+                *v1_key.invocation_uuid_ok_or()?,
+            ),
+            &status_v1.0.clone(),
+        );
+        storage.delete_key(&v1_key);
+        return Ok(Some(status_v1.0));
+    }
+
+    Ok(None)
+}
+
+fn get_invocation_status_v2<S: StorageAccess>(
```

Comment:
Can you add some explanation when to use `get_invocation_status` vs. `get_invocation_status_v2` vs. `try_migrate_on_read`? Could `try_migrate_on_read` and `get_invocation_status_v2` be merged? Right now, it seems that they are always used together.